pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/bestpractices/xml/UnnecessaryWarningSuppression.xml

```diff
@@ -0,0 +1,372 @@
+        </expected-messages>
+        <code><![CDATA[
+            public class Foo {
+                @SuppressWarnings({"PMD", "PMD.FakeRuleThatReportsIncrements"})
+                static void foo() {
+                    int i = 0;
+                    i++;
+                }
+                @SuppressWarnings({"PMD", "PMD.FakeRuleThatReportsIncrements"})
+                static void bar() {
+
+                }
+            }
+            ]]></code>
+    </test-code>
+    <test-code>
+        <description>annotation suppression (multiple parts, specific rule, all)</description>
+        <expected-problems>2</expected-problems>
+        <expected-linenumbers>2,7</expected-linenumbers>
+        <expected-messages>
+            <message>Unnecessary suppression "PMD.FakeRuleThatReportsIncrements" in @SuppressWarnings annotation</message>
+            <message>Unnecessary suppression "PMD.FakeRuleThatReportsIncrements" in @SuppressWarnings annotation</message>
+        </expected-messages>
+        <code><![CDATA[
+            public class Foo {
+                @SuppressWarnings({"all", "PMD.FakeRuleThatReportsIncrements"})
+                static void foo() {
+                    int i = 0;
+                    i++;
+                }
+                @SuppressWarnings({"all", "PMD.FakeRuleThatReportsIncrements"})
+                static void bar() {
+
+                }
+            }
+            ]]></code>
+    </test-code>
+    <test-code>
+        <description>annotation suppression (multiple parts, specific rule, all, inverted)</description>
+        <expected-problems>2</expected-problems>
+        <expected-linenumbers>2,7</expected-linenumbers>
+        <expected-messages>
+            <message>Unnecessary suppression "PMD.FakeRuleThatReportsIncrements" in @SuppressWarnings annotation</message>
+            <message>Unnecessary suppression "PMD.FakeRuleThatReportsIncrements" in @SuppressWarnings annotation</message>
+        </expected-messages>
+        <code><![CDATA[
+            public class Foo {
+                @SuppressWarnings({"PMD.FakeRuleThatReportsIncrements", "all"})
+                static void foo() {
+                    int i = 0;
+                    i++;
+                }
+                @SuppressWarnings({"PMD.FakeRuleThatReportsIncrements", "all"})
+                static void bar() {
+
+                }
+            }
+            ]]></code>
+    </test-code>
+    <test-code disabled="true">
+        <description>annotation suppression (multiple rules)</description>
+        <expected-problems>2</expected-problems>
+        <expected-linenumbers>2,8</expected-linenumbers>
+        <expected-messages>
+            <!--
+            TODO what is the correct behavior here? rule 1 is suppressed by "PMD.FakeRule...",
+               rule 2 is suppressed by "PMD", so both are in use. Should we report "PMD.FakeRule..."
+               because it is a subset of "PMD"?
```

Comment:
Maybe we can extend the message to mention, why it is unnecessary: Because it is already suppressed by "PMD" / "all".

Reply:
See #5773